Dockerfile.antidote

```diff
@@ -0,0 +1,113 @@
+ARG ELIXIR_VERSION=1.13.4
+ARG OTP_VERSION=24.3
+ARG DEBIAN_VERSION=bullseye-20210902-slim
+
+ARG BUILDER_IMAGE="hexpm/elixir:${ELIXIR_VERSION}-erlang-${OTP_VERSION}-debian-${DEBIAN_VERSION}"
+ARG RUNNER_IMAGE="debian:${DEBIAN_VERSION}"
+
+FROM ${BUILDER_IMAGE} AS builder
+
+LABEL maintainer="dev@vaxine.io"
+
+RUN apt-get update -y && apt-get install -y build-essential git curl \
+    && apt-get clean && rm -f /var/lib/apt/lists/*_*
+
+RUN mix local.hex --force                                                                                                                                                
+RUN mix local.rebar --force  
+
+WORKDIR /app
+
+ENV CI=true
+
+COPY rebar.config /app/
+COPY rebar.lock /app/
+
+RUN /root/.mix/rebar3 compile
+
+COPY Makefile /app/
```

Comment:
If you intend to support development usecase you should not copy those files, but instead mount the whole project instead and use the exact same targets used to compile locally.

Reply:
I actually kinda don't. This is a multistage build which builds an antidote release and packages that, I don't see a reason to mix that with local development usecase

Reply:
Because there is no need to rebuild an image from this step every time there is a change in a Makefile.

Reply:
Yeah, right now I'm actually calling rebar directly and we can drop copying the Makefile completely. I don't actually see any value in calling `make compile release` in Dockerfile, I feel like Dockerfile is declarative enough as far as build step descriptions go.

I'd be happy to be educated on the Makefile benefits though. From my point of view, Dockerfiles, especially optimized for production, commonly require running commands in combinations or in order that doesn't make a lot of sense outside the Dockerfile. For that reason, I don't see big benefits in putting "used-in-exactly-one-place" commands anywhere other than the place of usage.

Reply:
I initially put that step there because later step called `make rel` to build a release. Makefile didn't automatically pick up rebar3 from mix, so I went with the faster option of calling it directly, given we have only one actual command to call (`rebar3 release`)

Reply:
I also know that rebar is looked up at the top of the makefile; I just didn't want to adjust it while I was testing around :)

Reply:
One of the benefits of makefiles - single entry point, and de-duplication of the code around how to configure/compile.
With the approach that you have picked everytime we need to reorganize something or call rebar3 with slightly different parameters there is a need to go and fix in multiple places. While we could have just used `make compile` and `make release` for local development and if you want to build release in docker just call `make docker-compile` and `make docker-release` from your shell which would execute `make compile` and `make release` inside the container. So no matter which compilation you choose - there is a single path of commands that gonna be called.

This way you also decouple the specific environment being used (local, docker, prod, etc) you are building/running in from actual commands to execute those tasks. 

I understand that current build setup in the repo is still flacky, and we gonna adjust it later, so just general recommendations rather then a blocker for this PR.

Reply:
I kinda addressed this by
1. Using `make rel` in the dockerfile to actually build the release
2. I updated the `make docker-build` command to use this dockerfile. If you'd like me to use it in the `buildkite` pipeline file, then I'd appreciate your expertise on how to marry the local and CI commands

I cannot use `make` higher up instead of `rebar3 compile` when compiling dependencies, because I don't want to force depenency recompilation on Makefile changes (in my experience fetch & compile dependencies is the longest step of most builds for software such as ours)